### ISSUE TITLE
Add support for security plugin

### ISSUE THREAD

**Issue by 10000-ki** [opensearch-project/opensearch-java#991]
### Is your feature request related to a problem?

https://github.com/elastic/elasticsearch-java/tree/main/java-client/src/main/java/co/elastic/clients/elasticsearch/security

As you may know, elasticsearch-java-client supports  `SecurityClient`
This is a class that provides security-related convenience features.

Are there any plans to bring the Security Client to the OpenSearch Java Client?

**Comment by dblock** [comment]
Yes, we'd like support for the security plugin the client. The grand plan is to generate most of that code from spec (see #366), a lot of the spec for the security plugin has already been contributed to https://github.com/opensearch-project/opensearch-api-specification. We could definitely use some help!

**Comment by 10000-ki** [comment]
@dblock Thanks for explaining

https://github.com/opensearch-project/opensearch-java/pull/366

This work seems to have stalled over the past year.
I was wondering if there are any plans to continue development or if you would like someone to contribute.

**Comment by dblock** [comment]
> I was wondering if there are any plans to continue development or if you would like someone to contribute.

@Xtansia is actively working on the generator, however we would love help on the various specs
